### PR TITLE
Adding explicit codeVerifier parameter to the TokenRequest

### PR DESCRIPTION
**Pull request by ashward**
# Why

This adds an explicit codeVerifier parameter to the AccessTokenRequest in @EvanBacon's PR so that an Authorisation Code flow using PKCE can be carried out.

# How

I added the extra property, along with some tests.

# Test Plan

This is the code I've been using to test the flow:

```ts
    const codeResponse = await promptAsync({ useProxy: false });

    if (codeResponse['error'] === null && codeResponse['type'] === 'success') {
        if (discovery === null) {
            throw new Error('Auto discovery failed');
        }

        const tokenResponse = await exchangeCodeAsync(
            {
                clientId: config.clientId,
                code: codeResponse.params.code,
                redirectUri: config.redirectUri,
                codeVerifier: request?.codeVerifier
            },
            discovery
        );
    }
```

### CODE REVIEW

File: packages/expo-auth-session/src/TokenRequest.types.ts [hunk 1]
```diff
@@ -74,6 +74,12 @@ export interface AccessTokenRequestConfig extends TokenRequestConfig {
    * [Section 3.1.2](https://tools.ietf.org/html/rfc6749#section-3.1.2)
    */
   redirectUri: string;
+  /**
+   * If we are using PKCE then we include the original 'code_verifier' in the token request.
+   * 
+   * (https://tools.ietf.org/html/rfc7636)
```
Comment (EvanBacon) [inline packages/expo-auth-session/src/TokenRequest.types.ts:80]:
```suggestion
   * [Section 4.5](https://tools.ietf.org/html/rfc7636#section-4.5)
```

File: packages/expo-auth-session/src/TokenRequest.ts [hunk 2]
```diff
@@ -262,6 +265,10 @@ export class AccessTokenRequest extends TokenRequest<AccessTokenRequestConfig>
       queryBody.code = this.code;
     }
 
+    if(this.codeVerifier) {
```
Comment (EvanBacon) [inline packages/expo-auth-session/src/TokenRequest.ts:268]:
```suggestion
    if (this.codeVerifier) {
```

**Review by EvanBacon — CHANGES_REQUESTED** [review]
thanks, please also update the docs.

File: packages/expo-auth-session/src/TokenRequest.types.ts [hunk 3]
```diff
@@ -74,6 +74,12 @@ export interface AccessTokenRequestConfig extends TokenRequestConfig {
    * [Section 3.1.2](https://tools.ietf.org/html/rfc6749#section-3.1.2)
    */
   redirectUri: string;
+  /**
+   * If we are using PKCE then we include the original 'code_verifier' in the token request.
```
Comment (EvanBacon) [inline packages/expo-auth-session/src/TokenRequest.types.ts:78]: If possible, could you add a more informative doc block -- for instance I think people may not know what the original `code_verifer` is if they use the hooks API.

**Comment by raarts** [comment]
@EvanBacon, why was this closed? It seems this was never merged.

**Pull request closed without merging**